controllers/dspipeline_params.go

```diff
@@ -235,7 +235,11 @@ func (p *DSPAParams) SetupObjectParams(ctx context.Context, dsp *dspa.DataScienc
 		p.ObjectStorageConnection.Bucket = dsp.Spec.ObjectStorage.ExternalStorage.Bucket
 		p.ObjectStorageConnection.Host = dsp.Spec.ObjectStorage.ExternalStorage.Host
 		p.ObjectStorageConnection.Scheme = dsp.Spec.ObjectStorage.ExternalStorage.Scheme
-		p.ObjectStorageConnection.Secure = dsp.Spec.ObjectStorage.ExternalStorage.Secure
+		if p.ObjectStorageConnection.Scheme == "https" {
+			p.ObjectStorageConnection.Secure = true
+		} else {
+			p.ObjectStorageConnection.Secure = false
```

Comment:
Because the field exists in the CR spec, we want this field to take precedence if it's set within the CR, so your conditional needs to account for this (i.e. if `.Secure` is set, then use that value, otherwise infer from .Scheme). 

But this introduces a new issue, which is we do have a default value for `Secure`, which is [here](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/api/v1alpha1/dspipeline_types.go#L218). 

So we need to remove this default value from the crd spec, and always manually set it.

I believe this will introduce a new problem, which is if we don't set this value the default will become "false", since the `zero` value for bool is `false`. To get around this you will need to convert `Secure` to a type `*bool` and check for `nil` if it's unset.